src/cowrie/commands/wget.py

```diff
@@ -189,41 +185,55 @@ def success(self, data, outfile):
-            self.errorWrite('{} ERROR {}: {}\n'.format(time.strftime('%Y-%m-%d %T'), error.webStatus.decode(),
-                                                       error.webMessage.decode('utf8')))
+        # we need to handle 301 redirects separately
+        if hasattr(error, 'webStatus') and error.webStatus.decode() == '301':
+            self.errorWrite('{} {}\n'.format(error.webStatus.decode(), error.webMessage.decode()))
+            https_url = error.getErrorMessage().replace('301 Moved Permanently to ', '')
+            self.errorWrite('Location {} [following]\n'.format(https_url))
+
+            # do the download again with the https URL
+            self.deferred = self.download(https_url.encode('utf8'), self.outfile)
+            if self.deferred:
+                self.deferred.addCallback(self.success)
+                self.deferred.addErrback(self.error, https_url)
+            else:
+                self.exit()
         else:
-            self.errorWrite('{} ERROR 404: Not Found.\n'.format(time.strftime('%Y-%m-%d %T')))
+            if hasattr(error, 'getErrorMessage'):  # exceptions
+                errorMessage = error.getErrorMessage()
+                self.errorWrite(errorMessage + '\n')
+                # Real wget also adds this:
+            if hasattr(error, 'webStatus') and error.webStatus and hasattr(error, 'webMessage'):  # exceptions
+                self.errorWrite('{} ERROR {}: {}\n'.format(time.strftime('%Y-%m-%d %T'), error.webStatus.decode(),
+                                                           error.webMessage.decode('utf8')))
+            else:
+                self.errorWrite('{} ERROR 404: Not Found.\n'.format(time.strftime('%Y-%m-%d %T')))
 
-        # prevent cowrie from crashing if the terminal have been already destroyed
-        try:
-            self.protocol.logDispatch(eventid='cowrie.session.file_download.failed',
-                                      format='Attempt to download file(s) from URL (%(url)s) failed',
-                                      url=self.url)
-        except Exception:
-            pass
+            # prevent cowrie from crashing if the terminal have been already destroyed
+            try:
+                self.protocol.logDispatch(eventid='cowrie.session.file_download.failed',
+                                          format='Attempt to download file(s) from URL (%(url)s) failed',
+                                          url=self.url)
+            except Exception:
+                pass
 
-        self.exit()
+            self.exit()
 
 
 # From http://code.activestate.com/recipes/525493/
 class HTTPProgressDownloader(client.HTTPDownloader):
     def __init__(self, wget, fakeoutfile, url, outfile, headers=None):
-        client.HTTPDownloader.__init__(self, url, outfile, headers=headers, agent=b'Wget/1.11.4')
+        client.HTTPDownloader.__init__(self, url, outfile, headers=headers, agent=b'Wget/1.11.4', followRedirect=False)
```

Comment:
We need to do the redirect on our own. For some reason, twisted does not send the SNI automatically and the SSL handshake fails after automatic redirect.